interfaces/ITextToSpeech.h

```diff
@@ -0,0 +1,116 @@
+#ifndef __ITEXTTOSPEECH_H
+#define __ITEXTTOSPEECH_H
+
+#include "Module.h"
+
+namespace WPEFramework {
+namespace Exchange {
+
+    struct EXTERNAL ITextToSpeech : virtual public Core::IUnknown {
+        enum { ID = ID_TEXT_TO_SPEECH };
+
+        struct Configuration {
+        std::string ttsEndPoint;
+        std::string ttsEndPointSecured;
+        std::string language;
+        std::string voice;
+        uint8_t volume;
+        uint8_t rate;
+        };
+
+        enum SpeechState: uint8_t {
+        SPEECH_PENDING = 0,
+        SPEECH_IN_PROGRESS,
+        SPEECH_PAUSED,
+        SPEECH_NOT_FOUND
+        };
+
+        enum TTSErrorDetail: uint8_t {
+        TTS_OK = 0,
+        TTS_FAIL,
+        TTS_NOT_ENABLED,
+        TTS_INVALID_CONFIGURATION,
+        TTS_NO_ACCESS
+        };
+
+
+        struct INotification : virtual public Core::IUnknown {
+            enum { ID = ID_TEXT_TO_SPEECH_NOTIFICATION };
+
+            virtual ~INotification() {}
+            // Signal changes on the subscribed namespace
+            // @brief Notify TTS enabled/disabled
+            // @param state enabled/disabled
+            virtual void Enabled(const bool &state) = 0;
+            // @brief Notify change in voice used for speaking
+            // @param voice voice changed 
+            virtual void VoiceChanged(const string &voice) = 0;
+            // @brief Notify speechid based on the speech state(eg: start,pause,..etc)
+            // @param speechid id of the text
+            virtual void WillSpeak(const uint32_t &speechid) = 0;
```

Comment:
I have to admit, reviewing bottom - up :-) But also here do not use references on const scalar variables. See remarks below.